mempool/reactor.go

```diff
@@ -84,6 +132,7 @@ func (memR *MempoolReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte) {
 		if err != nil {
 			memR.Logger.Info("Could not check tx", "tx", msg.Tx, "err", err)
 		}
+		memR.MarkPeer(src, true, 1, p2p.Correct)
```

Comment:
should only happen if there's no error. or better yet, only if the asynchronous CheckTx call returns a success. maybe we can use the CheckTx callback argument for that

Reply:
Will do.